src/python/grpcio/grpc/_server.py

```diff
@@ -504,6 +539,7 @@ def _unary_response_in_pool(rpc_event, state, behavior, argument_thunk,
         if argument is not None:
             response, proceed = _call_behavior(rpc_event, state, behavior,
                                                argument, request_deserializer)
+            _maybe_request_compression(state, rpc_event)
```

Comment:
Do we need to call `_maybe_request_compression` here? Is receiving messages after sending the unary response allowed in stream-unary call?

Reply:
We definitely do need this here. Setting the `compression` attribute of the state object does nothing from core's perspective. `_maybe_request_compression` takes the value of the attribute and translates it into initial metadata interpretable by core.